### PR TITLE
bytecode: added support for self expression

### PR DESCRIPTION
**Pull request by hzahradnik**

### CODE REVIEW

File: src/bytecode/astgen.rs [hunk 1]
```diff
@@ -1410,7 +1437,7 @@ mod tests {
         let fct = code("fun f() -> Int { let x = 1; return x; }");
         let expected = vec![ConstInt(r(0), 1), RetInt(r(0))];
         assert_eq!(expected, fct.code());
-    }
+    }
```
Comment (dinfuehr) [inline src/bytecode/astgen.rs]: I think you added these spaces by mistake

File: src/bytecode/opcode.rs [hunk 2]
```diff
@@ -32,6 +32,7 @@ pub enum Bytecode {
     MovFloat(Register, Register),
     MovDouble(Register, Register),
     MovPtr(Register, Register),
+    MovSelf(Register, Register),
```
[the remark this answers is not in the document]
Comment (hzahradnik) [reply src/bytecode/opcode.rs]: You're right. I will change it.

File: src/bytecode/astgen.rs (hunk 1 above)
Comment (hzahradnik) [reply src/bytecode/astgen.rs]: Yep, I will remove them.

**Comment by hzahradnik** [comment]
I updated the PR to reflect the special case of primitives (Bool, Byte, Int, ...). These types are implemented as classes, but self should not return a pointer, but the value itself.

**Comment by dinfuehr** [comment]
Thanks!

**Pull request merged**